langcodes/__init__.py

```diff
@@ -217,6 +222,16 @@ def get(tag: str, normalize=True) -> 'Language':
         if (tag, normalize) in Language._PARSE_CACHE:
             return Language._PARSE_CACHE[tag, normalize]
 
+        if isinstance(tag, Language):
+            if not normalize:
+                # shortcut: we have the tag already
+                return tag
+            
+            # We might need to normalize this tag. Convert it back into a
+            # string tag, to cover all the edge cases of normalization in a
+            # way that we've already solved.
+            tag = tag.to_tag()
```

Comment:
It looks like this block wants to be *above* the previous if-block, so that you can use the _PARSE_CACHE for the `tag.to_tag()` string.